Lombiq.Tests.UI/MonkeyTesting/UrlFilters/NotStartsWithMonkeyTestingUrlFilter.cs

```diff
@@ -17,5 +16,5 @@ public class NotStartsWithMonkeyTestingUrlFilter : IMonkeyTestingUrlFilter
         _relativeUrlNotStartsWith = relativeUrlNotStartsWith;
 
     public bool AllowUrl(UITestContext context, Uri url) =>
-       !_relativeUrlNotStartsWith.Any(filterUrl => url.PathAndQuery.StartsWithOrdinalIgnoreCase(filterUrl));
+       !Array.Exists(_relativeUrlNotStartsWith, url.PathAndQuery.StartsWithOrdinalIgnoreCase);
```

Comment:
Looking at this (again), I shudder due to the inconsistent code style we'll be using:

![image](https://github.com/Lombiq/UI-Testing-Toolbox/assets/164360/207df16f-7209-47ab-a73c-1849a9f1b665)

I'd be heavily in favor of providing the necessary methods (or all) as extension methods on the the `Array` class and use them as instance methods instead of these unintuitive static invocations.

Reply:
It's not too intuitive, but [the rule docs](https://rules.sonarsource.com/csharp/RSPEC-6605) suggests it, so it's not a big leap to figure it out. I guess there's some reason why there's no such extension in `System.Linq`. Anyway, I added them.

I really don't have in me to go through the whole of `Array` to add everything we might ever need, but I added the ones we already use.

Reply:
I think you've covered the most important ones. We can always add more later.